nats.go

```diff
@@ -3740,8 +3739,13 @@ func (s *Subscription) processNextMsgDelivered(msg *Msg) error {
 
 	// In case this is a JetStream message and in pull mode
 	// then check whether it is an JS API error.
-	if jsi != nil && jsi.pull > 0 && len(msg.Data) == 0 && msg.Header.Get(statusHdr) == noResponders {
-		return ErrNoResponders
+	if jsi != nil && jsi.pull && len(msg.Data) == 0 {
+		switch msg.Header.Get(statusHdr) {
+		case noResponders:
+			return ErrNoResponders
+		case "400", "404", "408", "409":
```

Comment:
Convert this case for > 400 to future proof this?  (either convert to int or test the first char to '4' or '5')...  Alternatively, could add a default that results in an error.  wdyt?